File: queue-service/service/queue.go
package service

import (
	"queue-service/models"
	"queue-service/repository"
)

type QueueService struct {
	repo *repository.RedisRepository
}

func NewQueueService(repo *repository.RedisRepository) *QueueService {
	return &QueueService{
		repo: repo,
	}
}

func (s *QueueService) AddToQueue(orderID, userID int) (*models.QueueItem, error) {
	return s.repo.AddToQueue(orderID, userID)
}

func (s *QueueService) GetQueueItem(orderID int) (*models.QueueItem, error) {
	return s.repo.GetQueueItem(orderID)
}

func (s *QueueService) GetActiveQueue() ([]models.QueueItem, error) {
	return s.repo.GetActiveQueue()
}

func (s *QueueService) UpdateQueueStatus(orderID int, status string) error {
	return s.repo.UpdateQueueStatus(orderID, status)
}

func (s *QueueService) RemoveFromQueue(orderID int) error {
	return s.repo.RemoveFromQueue(orderID)
}

func (s *QueueService) GetQueueStats() (*models.QueueStats, error) {
	return s.repo.GetQueueStats()
}

File: frontend/lib/queue.ts
import { queueApi } from './api';
import { QueueItem, QueueStats } from './types';

export const queueService = {
  addToQueue: async (orderData: { order_id: number; user_id: number }): Promise<QueueItem> => {
    const response = await queueApi.post('/api/queue/', orderData);
    return response.data;
  },

  getQueueItem: async (orderId: number): Promise<QueueItem> => {
    const response = await queueApi.get(`/api/queue/order/${orderId}`);
    return response.data;
  },

  getActiveQueue: async (): Promise<QueueItem[]> => {
    const response = await queueApi.get('/api/queue/active');
    return response.data;
  },

  updateQueueStatus: async (orderId: number, status: string): Promise<QueueItem> => {
    const response = await queueApi.put(`/api/queue/order/${orderId}`, { status });
    return response.data;
  },

  removeFromQueue: async (orderId: number): Promise<void> => {
    await queueApi.delete(`/api/queue/order/${orderId}`);
  },

  getQueueStats: async (): Promise<QueueStats> => {
    const response = await queueApi.get('/api/queue/stats');
    return response.data;
  },
};